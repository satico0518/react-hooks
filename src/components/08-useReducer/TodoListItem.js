import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({
  todo,
  i,
  handleComplete,
  handleEdit,
  handleDelete,
}) => {
  return (
    <li className="list-group-item" key={todo.id}>
      <div className="row">
        <div className="col-9">
          <p
            style={{ cursor: 'pointer' }}
            className={`${todo.done && 'complete'}`}
            onClick={() => handleComplete(todo.id)}
          >
            {i + 1}. {todo.todo}
          </p>
        </div>
        <div className="col-3">
          <div className="text-right">
            <button
              className="btn btn-warning mr-2"
              onClick={() => handleEdit(todo)}
            >
              Edit
            </button>
            <button
              className="btn btn-danger"
              onClick={() => handleDelete(todo.id)}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </li>
  );
};

TodoListItem.propTypes = {
  todo: PropTypes.object.isRequired,
  handleComplete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoListItem;
