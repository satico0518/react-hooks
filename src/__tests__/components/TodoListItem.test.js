import React from 'react';

const { shallow } = require('enzyme');
import '@testing-library/jest-dom';
import TodoListItem from '../../components/08-useReducer/TodoListItem';

describe('TodoListItem', () => {
  const todo = { id: 1, todo: 'Make bed', done: false };
  const i = 0;
  const handleComplete = jest.fn();
  const handleEdit = jest.fn();
  const handleDelete = jest.fn();
  test('should match snapshot', () => {
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={i}
        handleComplete={handleComplete}
        handleEdit={handleEdit}
        handleDelete={handleDelete}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should render todo ok', () => {
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={i}
        handleComplete={handleComplete}
        handleEdit={handleEdit}
        handleDelete={handleDelete}
      />
    );
    expect(wrapper.find('p').text()).toBe(`${i + 1}. ${todo.todo}`);
  });
  test('should call handleComplete', () => {
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={i}
        handleComplete={handleComplete}
        handleEdit={handleEdit}
        handleDelete={handleDelete}
      />
    );
    wrapper.find('p').simulate('click');
    expect(handleComplete).toHaveBeenCalledTimes(1);
    expect(handleComplete).toHaveBeenCalledWith(todo.id);
  });
  test('should call handleEdit', () => {
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={i}
        handleComplete={handleComplete}
        handleEdit={handleEdit}
        handleDelete={handleDelete}
      />
    );
    wrapper.find('.btn-warning').simulate('click');
    expect(handleEdit).toHaveBeenCalledTimes(1);
    expect(handleEdit).toHaveBeenCalledWith(todo);
  });
  test('should call handleDelete', () => {
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={i}
        handleComplete={handleComplete}
        handleEdit={handleEdit}
        handleDelete={handleDelete}
      />
    );
    wrapper.find('.btn-danger').simulate('click');
    expect(handleDelete).toHaveBeenCalledTimes(1);
    expect(handleDelete).toHaveBeenCalledWith(todo.id);
  });
});
