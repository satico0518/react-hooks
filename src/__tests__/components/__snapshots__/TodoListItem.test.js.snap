// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem should match snapshot 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <div
    className="row"
  >
    <div
      className="col-9"
    >
      <p
        className="false"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        1
        . 
        Make bed
      </p>
    </div>
    <div
      className="col-3"
    >
      <div
        className="text-right"
      >
        <button
          className="btn btn-warning mr-2"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="btn btn-danger"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</li>
`;
